core/lane.py

```diff
@@ -484,10 +486,14 @@ def _values_from_request(cls, config, get_argument, get_header):
         g = Facets.DISTRIBUTOR_FACETS_GROUP_NAME
         distributor = get_argument(g, cls.default_facet(config, g))
         distributor_facets = cls.available_facets(config, g)
-        if distributor and distributor not in distributor_facets:
+        if (
+            distributor
+            and distributor != "All"
```

Comment:
The default facets for both distributor and collectioName have the "All" value [[source](https://github.com/ThePalaceProject/circulation/blob/addac46fa66a7714fa0a183d7cf10ec6c758940b/core/facets.py#L124)], why would this special case be required?

Reply:
While debugging this through the [lane search route](https://github.com/ThePalaceProject/circulation/blob/74f8668d1b2e15cec303248132cbcdd7a2fa2b5b/api/routes.py#L456-L463) with data from our dev CM, the "All" value was not present in `distributor_facets`, which looked like this:
```
[None, 'Torrossa Test', 'Axis 360', 'Overdrive', 'BiblioBoard', 'Failing Feed Test', ...]
```

Reply:
Ok I understand, the dymanic facet names method did not account for "All". This would work.